<launch>

	<!--remote control -->	
	  <arg name="map_file" default="$(find rc_bringup)/map/laba_209.yaml"/>
        <node pkg="rc_bringup" name="encoders" type="read_from_encoders.py" respawn="true" output="screen"/>
        <!--remote controll -->
        <node pkg="rc_bringup" name="rc_node" type="rc_control.py" respawn="true" output="screen">
	<param name="cmd_vel" value="cmd_vel"/>
	<param name="pwm_topic" value="pwm"/>
	<param name="drive_topic" value="ackermann_cmd"/>
	<param name="max_vel" value="2.5"/>
	<param name="min_vel" value="-2.5"/>
	<param name="wheelbase" value="0.28"/>
	<param name="max_steering_angle" value="25.0"/>
	<param name="servo_pin" value="22"/>
	<param name="middle_servo" value="1550"/>
	<param name="servo_offset" value="47"/>    
	<param name="motor_pin" value="4"/>
	<param name="middle_motor" value="1550"/>
        <param name="revers_servo" value="True"/>
	</node>

       
        <!--<node pkg="rc_bringup" name="laser2PointCloud" type="laser2pc.py" respawn="true" output="screen"/> -->
	<!--<node pkg="rc_bringup" name="to_point_cloud" type="to_point_cloud.py" respawn="true" output="screen"/> -->

        <!--pose controller -->
<!--
<node pkg="rc_bringup" name="rc_pose_controller" type="pose_controller_with_reg_func4_5_dif_coeff.py" respawn="true" output="screen">
       
                <param name="vel_topic" value="velocity"/>
                
                <param name="pose_topic" value="pose"/>
                <param name="goal_topic" value="goal"/>
                <param name="local_goal_topic" value="local_goal"/>

                <param name="max_vel" value="0.6"/>
                <param name="min_vel" value="-2.5"/>
                <param name="max_angle" value="25.0"/>
                <param name="goal_tolerance" value="0.2"/>

                <param name="kP_pose" value="3.5"/>
                <param name="kI_pose" value="0.0"/>
                <param name="kD_pose" value="0.001"/>
                <param name="kP_course" value="2.4"/>
                <param name="kI_course" value="0.1"/>
                <param name="kD_course" value="0.4"/>
        </node>
-->
        <!--
        <node pkg="rc_bringup" name="unstable" type="unstable_planner.py" respawn="true" output="screen"/>
                 <param name="goal_topic" value="goal"/>
                <param name="goal_tolerance" value="0.2"/>
        </node>
-->
       
        <!--<node pkg="rc_bringup" name="rc_route_generator" type="route_generation.py" respawn="true"/> -->
        
      <!--  <node pkg="rc_bringup" name="rc_pot_fields" type="pose_controller_with_reg_func4_5_dif_coeff.py" respawn="true" output="screen"/>
      -->

</launch>
